util/chunk/chunk_util.go

```diff
@@ -13,20 +13,25 @@
 
 package chunk
 
+import "github.com/pingcap/errors"
+
 // CopySelectedJoinRows copies the selected joined rows from the source Chunk
 // to the destination Chunk.
 // Return true if at least one joined row was selected.
 //
 // NOTE: All the outer rows in the source Chunk should be the same.
-func CopySelectedJoinRows(src *Chunk, innerColOffset, outerColOffset int, selected []bool, dst *Chunk) bool {
+func CopySelectedJoinRows(src *Chunk, innerColOffset, outerColOffset int, selected []bool, dst *Chunk) (bool, error) {
 	if src.NumRows() == 0 {
-		return false
+		return false, nil
+	}
+	if src.sel != nil || dst.sel != nil {
+		return false, errors.Trace(ErrSelShouldBeNil)
```

Comment:
How about calling `errors.New()` here? Just make `ErrSelShouldBeNil` an error message. For example:
```go
var msgErrSelNotNil = "The selection vector of Chunk is not nil. Please file a bug to the TiDB Team"

...

return false, errors.New(msgErrSelNotNil)
```